### PR TITLE
chore(ci): remove duplicated workflows in `main-cron`

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Seems like a bunch of workflows were duplicated in `main-cron`. Perhaps changes from #12233 and https://github.com/risingwavelabs/risingwave/pull/12232 were duplicated because of slightly different test headings and different orderings.

```
> cat ci/workflows/main-cron.yml | grep label: | sort | sed -E 's/ \(release\)//' | uniq -c | grep '[2-9]  '
   2   - label: "S3 source check on AWS (json parser)"
   2   - label: "connector node integration test Java {{matrix.java_version}}"
   2   - label: "e2e java-binding test"
   2   - label: "end-to-end iceberg sink test"
   2   - label: "end-to-end iceberg sink v2 test"
   2   - label: "end-to-end test (parallel, in-memory)"
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -478,7 +442,7 @@ steps:
           config: ci/docker-compose.yml
           mount-buildkite-agent: true
       - ./ci/plugins/upload-failure-logs
-    timeout_in_minutes: 21
+    timeout_in_minutes: 25
```
Comment (kwannoel) [inline ci/workflows/main-cron.yml:445]: Timeout of standalone test bumped also.

Comment (kwannoel) [reply ci/workflows/main-cron.yml:445]: Since it is TLE in `main-cron`

**Review by fuyufjh — APPROVED** [review]
Thanks!! ❤️

**Comment by kwannoel** [comment]
Screenshot 2023-09-29 at 11 48 31 AM

This PR seems to have resolved most of the issues. Pr fuzz test should be fixed by #12592 . Looking at pulsar test.

**Pull request merged**